upup/pkg/fi/nodeup/nodetasks/bindmount_test.go

```diff
@@ -0,0 +1,253 @@
+		}
+		if len(g.executor.Commands) != 0 {
+			t.Errorf("not all expected commands were called: %s", g.executor.Commands)
+		}
+	}
+}
+
+func TestBindMountDependencies(t *testing.T) {
+	containerizedMounterHome := "/containerized_mounter"
+
+	grid := []struct {
+		parent fi.Task
+		child  fi.Task
+	}{
+		{
+			parent: &MountDiskTask{
+				Mountpoint: "/",
+			},
+			child: &BindMount{
+				Source:     containerizedMounterHome,
+				Mountpoint: containerizedMounterHome,
+				Options:    []string{"exec"},
+			},
+		},
+		{
+			parent: &File{
+				Path: containerizedMounterHome,
+				Type: FileType_Directory,
+			},
+			child: &BindMount{
+				Source:     containerizedMounterHome,
+				Mountpoint: containerizedMounterHome,
+				Options:    []string{"exec"},
+			},
+		},
+		{
+			parent: &Archive{
+				TargetDir: containerizedMounterHome,
+			},
+			child: &BindMount{
+				Source:     containerizedMounterHome,
+				Mountpoint: containerizedMounterHome,
+				Options:    []string{"exec"},
+			},
+		},
+		{
+			parent: &BindMount{
+				Source:     containerizedMounterHome,
+				Mountpoint: containerizedMounterHome,
+				Options:    []string{"exec"},
+			},
+			child: &BindMount{
+				Source:     "/var/lib/kubelet/",
+				Mountpoint: path.Join(containerizedMounterHome, "rootfs/var/lib/kubelet"),
+				Options:    []string{"rshared"},
+				Recursive:  true,
+			},
+		},
+		{
+			parent: &BindMount{
+				Source:     containerizedMounterHome,
+				Mountpoint: containerizedMounterHome,
+				Options:    []string{"exec"},
+			},
+			child: &BindMount{
+				Source:     "/proc",
+				Mountpoint: path.Join(containerizedMounterHome, "rootfs/proc"),
+				Options:    []string{"ro"},
+			},
+		},
+		{
+			parent: &BindMount{
+				Source:     containerizedMounterHome,
+				Mountpoint: containerizedMounterHome,
+				Options:    []string{"exec"},
+			},
+			child: &BindMount{
+				Source:     "/dev",
+				Mountpoint: path.Join(containerizedMounterHome, "rootfs/dev"),
+				Options:    []string{"ro"},
+			},
+		},
+		{
+			parent: &BindMount{
+				Source:     containerizedMounterHome,
+				Mountpoint: containerizedMounterHome,
+				Options:    []string{"exec"},
+			},
+			child: &BindMount{
+				Source:     "/etc/resolv.conf",
+				Mountpoint: path.Join(containerizedMounterHome, "rootfs/etc/resolv.conf"),
+				Options:    []string{"ro"},
+			},
+		},
+	}
+
+	for _, g := range grid {
+		allTasks := make(map[string]fi.Task)
+		allTasks["parent"] = g.parent
+		allTasks["child"] = g.child
+
+		deps := g.parent.(fi.HasDependencies).GetDependencies(allTasks)
+		if len(deps) != 0 {
+			t.Errorf("found unexpected dependencies for parent: %v %v", g.parent, deps)
+		}
+
+		childDeps := g.child.(fi.HasDependencies).GetDependencies(allTasks)
+		if len(childDeps) != 1 {
+			t.Errorf("found unexpected dependencies for child: %v %v", g.child, childDeps)
+		}
+	}
+}
+
+// MockExecutor is a mock implementation of Executor
+type MockExecutor struct {
```

Comment:
we should start a mock under /pkg :)  We should be using this in other places.

Reply:
RIght, I was going to move it when the second use arose :-)